### PR TITLE
Move update_data method from Entity to EntitySet

### LINKED ISSUE #1367: Move `update_data` from Entity to EntitySet
Featuretools should provide an interface for users to update the data that makes up an "entity" in an EntitySet. This is currently done via a method on the `ft.Entity` object, but Woodwork dataframes do not have a corresponding method. To resolve this problem, we can move the `update_data` method from `Entity` to `EntitySet`.

### PR DESCRIPTION
**Pull request by jeff-hernandez**
Closes #1367 by making `update_data` an entity set method

### CODE REVIEW

**Comment by thehomebrewnerd** [comment]
@jeff-hernandez Couple things here. I think we probably want to submit this PR against the `woodwork-integration` branch instead of `main`. Can you make that update and confirm that this branch is up to date with the `woodwork-integration` branch?

With that change, I think you might have a conflict in release notes to clean up. Also, in the release notes, please add this as a breaking change and include as sample code block that demonstrates the new API as we have done with the other breaking changes in the `woodwork-integration` branch.

After those updates are made, I can take a closer look, but based on a quick scan, things are looking pretty good.

**Comment by rwedge** [comment]
Given this PR is woodowork-aware, what if we changed the method name to `update_dataframe`?  `update_data` seems a little too ambiguous as an entityset method, like it be applied to multiple dataframes at once

**Comment by jeff-hernandez** [comment]
@rwedge that makes sense

**Comment by thehomebrewnerd** [comment]
> Given this PR is woodowork-aware, what if we changed the method name to `update_dataframe`? `update_data` seems a little too ambiguous as an entityset method, like it be applied to multiple dataframes at once

Seems like a good change to me

**Review by thehomebrewnerd — APPROVED** [review]
Just one really minor wording suggestion, but otherwise looks good to me.

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -12,6 +12,7 @@ Future Release
         * Move ``set_secondary_time_index`` method from ``Entity`` to ``EntitySet`` (:pr:`1280`)
         * Refactor Relationship creation process (:pr:`1370`)
         * Add auto assign bot on GitHub (:pr:`1380`)
+        * Remove ``Entity.update_data`` and add ``EntitySet.update_dataframe`` (:pr:`1398`)
```
[the remark this answers is not in the document]
Comment (jeff-hernandez) [reply docs/source/release_notes.rst]: ✅ a6e6aef8

**Pull request merged**